src/components/dapps/ModalCreateDapps.vue

```diff
@@ -284,32 +306,371 @@
+
+    const {
+      fileRef: wasmFromFile,
+      setFile: setWasmFile,
+      validate: isWasmFromFileValid,
+    } = useFile({
+      onChange: onChangeAbi,
+      onRemove: onRemoveAbi,
+      validate: (file) =>
+        file?.data.subarray(0, 4).toString() === '0,97,115,109',
+    });
+
+    const wasm = ref();
+    const isWasmValid = ref(false);
+
+    const onDropFile = (fileState: FileState) => {
+      setWasmFile(fileState);
+    };
+
+    watch(abi, () => {
+      console.log('abc', abi.value);
+      console.log('wasm', abi.value?.project.source.wasm);
+      console.log('constructor', abi.value?.constructors);
+      if (abi.value && isWasm(abi.value.project.source.wasm)) {
+        wasm.value = abi.value.project.source.wasm;
+        isWasmValid.value = true;
+
+        // if (currentName.current === '') {
+        //   setName(`${abi.value.project.contract.name.toString()}.contract`);
+        // }
+        return;
+      }
+
+      if (wasmFromFile.value && isWasmFromFileValid) {
+        wasm.value = compactAddLength(wasmFromFile.value.data);
+        isWasmValid.value = true;
+
+        return;
+      }
+
+      wasm.value = null;
+      isWasmValid.value = false;
+    });
+
+    const upload = async () => {
+      if (
+        formData.projectName === '' ||
+        toAddress.value === '' ||
+        wasm.value === null
+      ) {
+        store.dispatch(ActionTypes.SHOW_ALERT_MSG, {
+          msg: `Please check fields again`,
+          alertType: 'error',
+        });
+        return;
+      }
+
+      const abiData = abi.value as Abi | AnyJson;
+
+      console.log('s', abiData);
+      console.log('w', wasm.value);
+
+      const code = new CodePromise(api?.value, abiData, wasm.value);
+      let uploadTx: SubmittableExtrinsic<'promise'> | null = null;
+
+      try {
+        //should be changable
+        // const unit_d = 3;
+        // const decimal = 12;
+        // const endowment = bnToBn(
+        //   plasmUtils.reduceDenomToBalance(27, unit_d, decimal)
+        // );
+        // const weight = new BN(200000000000);
+        ///
+        console.log('endowment', formData.endowment);
+        console.log('weight', formData.weight);
+
+        console.log('code', code);
+        console.log('method', code.tx);
+
+        const constructorIndex = 0;
+        const params = abi?.value?.constructors[constructorIndex].args;
+        console.log('params', params);
+        const pvalues = params?.reduce(
+          (result: RawParams, param): RawParams => [
+            ...result,
+            createValue(abi?.value?.registry, param),
+          ],
+          []
+        );
+        const arrValues: any = pvalues?.map(({ value }) => value);
+        console.log('v', arrValues);
+
+        uploadTx =
+          code &&
+          abi.value?.constructors[constructorIndex].method &&
+          formData.endowment
+            ? code.tx[abi.value?.constructors[constructorIndex].method](
```

Comment:
It may not be good to parse the contract methods this way. I recommend using a structure like this https://github.com/polkadot-js/apps/blob/0ece312038b78d0191a1747be78afb9342a64e17/packages/page-contracts/src/Codes/Upload.tsx#L185-L197

In short, to create the contract constructor parameter, you need `abi.constructors[constructorIndex].args` and `abi.registry`